cpp/CMakeLists.txt

```diff
@@ -55,6 +55,13 @@ option(RAFT_COMPILE_LIBRARIES "Enable building raft shared library instantiation
 option(RAFT_COMPILE_NN_LIBRARY "Enable building raft nearest neighbors shared library instantiations" OFF)
 option(RAFT_COMPILE_DIST_LIBRARY "Enable building raft distant shared library instantiations" OFF)
 option(RAFT_ENABLE_NN_DEPENDENCIES "Search for raft::nn dependencies like faiss" ${RAFT_COMPILE_LIBRARIES})
+
+option(RAFT_ENABLE_cuco_DEPENDENCY "Enable cuCollections dependency" ON)
```

Comment:
When is it recommended to use this flag? It doesn't seem to be used currently in a few places (like line 150) to disable cuco::cuco

Reply:
Was this comment perhaps made on an older version of the changes in this PR? The Line 153 contains the following, which will add cuco to link library dependencies when it's been found in `find_package`:
`$<$<BOOL:${cuco_ADDED}>:cuco::cuco>`. The goal here is to allow upstream projects to add it if needed but allow it to not be added if `find_package` was never invoked for it.

Reply:
Yeah, when seeing the file github showed me an older version in the branch of the PR that didn't have that change, so now it makes sense